packages/backend/src/helpers/get-app.ts

```diff
@@ -40,18 +40,18 @@ const getApp = (appKey: string, stripFuncs = true) => {
   return appData
 }
 
-const chooseConnectionStep = {
-  key: 'chooseConnection',
-  name: 'Choose connection',
-}
-
-const testStep = (stepType: 'trigger' | 'action') => {
+const chooseConnectionStep = (connectionStepLabel?: string) => {
```

Comment:
i think we can omit the backend changes, since the frontend always needs to account for per-action/per-trigger labels. So this is kinda duplicating logic and spreads complexity across both backend and frontend.

I would instead consolidate all the complexity inside the frontend by making it handle all these label changes. For example, I could change `FlowStep` to something like this:
```
          {substeps?.length > 0 &&
            substeps.map((substep: ISubstep, index: number) => (
              <Fragment key={`${substep?.name}-${index}`}>
                {substep.key === 'chooseConnection' && app && (
                  <ChooseConnectionSubstep
                    expanded={currentSubstep === index + 1}
                    substep={substep}
                    onExpand={() => toggleSubstep(index + 1)}
                    onCollapse={() => toggleSubstep(index + 1)}
                    onSubmit={expandNextStep}
                    onChange={handleChange}
                    application={app}
                    step={step}
                    label={app.connectionSetupLabel ?? substep.name} // NEW
                  />
                )}
                {substep.key === 'testStep' && (
                  <TestSubstep
                    expanded={currentSubstep === index + 1}
                    substep={substep}
                    onExpand={() => toggleSubstep(index + 1)}
                    onCollapse={() => toggleSubstep(index + 1)}
                    onChange={handleChange}
                    onContinue={onContinue}
                    step={step}
                    selectedActionOrTrigger={selectedActionOrTrigger}
                  />
                )}
                {substep.key &&
                  ['chooseConnection', 'testStep'].includes(substep.key) ===
                    false && (
                    <FlowSubstep
                      expanded={currentSubstep === index + 1}
                      substep={substep}
                      onExpand={() => toggleSubstep(index + 1)}
                      onCollapse={() => toggleSubstep(index + 1)}
                      onSubmit={expandNextStep}
                      onChange={handleChange}
                      step={step}
                      settingsLabel={selectedActionOrTrigger?.settingsStepLabel ?? app.settingsStepLabel ?? substep.name} // NEW
                    />
                  )}
              </Fragment>
```

Reply:
hmm i get what you mean, but it seems like on the frontend, there isn't much logic for the labels also? not sure if it was a bad practice with the initial code